File: cpp/1343-number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold.cpp
/*
Using Sliding window.
First we calculate sum of first k elements in vector then we check if the average is greater than or equal to the threshold.
If the condition is met then we append the res variable.
Then we maintain the k size window and traverse over the vector and check for the condition.

Since we are traversing only once
T.C -> O(N)
S.c -> O(1)
*/
class Solution {
public:
    int numOfSubarrays(vector<int>& arr, int k, int threshold) {
        int sum = 0;
        int n = arr.size();
        for(int i=0;i<k;i++){
            sum += arr[i];
        }
        int left = 0;
        int right = k;
        int res = 0;
        if(sum/k >= threshold) res++;
        while(right < n){
            sum -= arr[left++];
            sum += arr[right++];
            if(sum/k >= threshold) res++;
        }
        return res;
    }
};

File: java/1343-number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold.java
class Solution {
    public int numOfSubarrays(int[] arr, int k, int threshold) {
        int res = 0, currSum = 0;

        for (int i = 0; i < k - 1; i++) {
            currSum += arr[i];
        }

        for (int l = 0; l < arr.length - k + 1; l++) {
            currSum += arr[l + k - 1];
            if ((currSum / k) >= threshold) {
                res++;
            }
            currSum -= arr[l];
        }
        return res;
    }
}
